subprojects/scala/src/main/java/org/gradle/language/scala/tasks/AbstractScalaCompile.java

```diff
@@ -182,41 +176,23 @@ protected ScalaJavaJointCompileSpec createSpec() {
     }
 
     private void configureExecutable(MinimalJavaCompilerDaemonForkOptions forkOptions) {
-        if (javaLauncher.isPresent()) {
-            forkOptions.setExecutable(javaLauncher.get().getExecutablePath().getAsFile().getAbsolutePath());
-        } else {
-            forkOptions.setExecutable(Jvm.current().getJavaExecutable().getAbsolutePath());
-        }
+        String executable = getJavaLauncher().get().getExecutablePath().getAsFile().getAbsolutePath();
+        forkOptions.setExecutable(executable);
```

Comment:
❌ Instead of mutating `forkOptions` here, shouldn't we set `javaHome` on `DefaultForkingScalaJavaJointCompileSpec` instead? It seems like the factory `DefaultScalaJavaJointCompileSpecFactory` is currently doing strange things: It should only create the `DefaultForkingScalaJavaJointCompileSpec`, since that is all the task can do now, right?

Then we can get rid of `DefaultCommandLineScalaJavaJointCompileSpec` I suppose.

Or maybe we could add the executable directly to `DefaultScalaJavaJointCompileSpec` and get rid of the two subclasses.

Reply:
That's a good suggestion, given then Java-related selection process for the spec get ignored by our Scala compiler wrapper in the end.

I went all the way to keep only `DefaultScalaJavaJointCompileSpec` and added the executable there. This simplified the code significantly in that part.